src/Proto3/Wire/Decode.hs

```diff
@@ -178,12 +178,20 @@ getFields = do
     unless e $ fail "Encountered bytes that aren't valid key/value pairs."
     return (toMap keyvals)
 
-toMap :: (Eq k, Hashable k, Ord k) => [(k, v)] -> M.Map k (Seq v)
+-- | Convert key-value pairs to a map of keys to a sequence of values with that
+-- key, in their original occurrence order.
+--
+-- >>> toMap ([("k1", 3),("k2", 4),("k1", 6)] :: [(String,Int)])
```

Comment:
Could you add this module to the set of modules to `doctest`?  I don't think this module is being tested yet